plugin/service/storage/plugin.go

```diff
@@ -292,9 +282,13 @@ func (sp *StoragePlugin) ValidatePermissions(ctx context.Context, req *pb.Valida
 		return nil, err
 	}
 
-	err = ensureServiceAccount(ctx, sa, sec)
-	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "failed to ensure service account: %v", err)
+	if !sec.LastRotatedTime.IsZero() {
+		// since these are existing secrets, we only want to validate the
+		// service account if the credentials have been rotated previously
+		err = ensureServiceAccount(ctx, sa, sec)
```

Comment:
Is this true? If it is the first time we are rotating credentials and its not a service account wont the rotation fail? My understanding was we cannot rotate if its not a service account, does this change then make it so we fail to rotate instead of validating the service account and fail on that? If that is the case is this check then just redundant?